doc/Coding Guidelines.md

```diff
@@ -0,0 +1,107 @@
+# Vue Storefront Coding Guidelines - DRAFT
+
+Hello Extension / Theme / Core developer! This is just a short guide to align the code style among the core modules and all the extensions and thems.
+
+## General guidelines
+In general we're using kind of mix of: [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript) and [JavaScript Standard Style](https://github.com/feross/standard/blob/master/RULES.md#javascript-standard-style). There are a eslint and vue-eslint configured to enforce the style. Please do use the same [configs in Your modules](https://github.com/DivanteLtd/vue-storefront/blob/master/.eslintrc.js). 
+
+## ES2015
+We do preffer to use ES2015 constructions in the code - and there is a babel compiler set to allow us to do so. So use the ES modules instead of `require` and feel free to use other fancy structures.
+
+## Components
+Components should be created as [Single File Vue components](https://vuejs.org/v2/guide/single-file-components.html) and names using `CamelCase.vue`.
+All the component properties should be named using `camelCase` as well.
+All the component data fields should be `camelCased`.
+
+Theme catalog structure is well described in here: [Working with themes](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/themes/Working%20with%20themes.md)
+
+## Extensions
+For the reference of how extension can be structured please take a look at [template](https://github.com/DivanteLtd/vue-storefront/tree/develop/src/extensions/template).
+Check the [Working with extensions](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/extensions/Working%20with%20extensions.md) for more info.
+
+It's something like this:
+
+```
+|____store.js
+|____index.js
+|____components
+| |____ComponentExample.vue
+|____router.js
+|____package.json
+|____pages
+| |____PageExample.vue
```

Comment:
I'd suggest to keep folders on top and then files, both in alphabetical order.

Reply:
I think we should just link to this folder instead of writing down the structure. When it changes it will require extra step to be up-to-date in docs ;)